[AnnabelTing] iP

AnnabelTing opened the pull request:
### ChickyChat
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))
ChickyChat frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use
All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:
And it is **FREE**!
Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
***
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

AnnabelTing committed:
Level-0

AnnabelTing committed:
Level-1

AnnabelTing committed:
Level-2

AnnabelTing committed:
Level-3

AnnabelTing committed:
Level-4

AnnabelTing committed:
A-TextUiTesting

AnnabelTing committed:
Level-5

AnnabelTing committed:
Level-6

AnnabelTing committed:
Level-7

AnnabelTing committed:
no message

AnnabelTing committed:
no message

AnnabelTing committed:
Revert "no message"

This reverts commit ad26b740f659eb93c7e7991c385cfdacb3a2e9f4.

AnnabelTing committed:
Merge branch 'branch-Level-7'

AnnabelTing committed:
Level-8

AnnabelTing committed:
A-MoreOOP

AnnabelTing committed:
A-Packages

AnnabelTing committed:
Merge branch 'add-gradle-support'

AnnabelTing committed:
A-Gradle

kayabuttertoastt commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,10 @@
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+public abstract class Command {
```
I like the nice usage of inheritance here. It  makes it much more extensible!

kayabuttertoastt commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,25 @@
+import duke.ui.Ui;
+
+public class MarkCommand extends Command{
+    int taskNumber;
```
It would be better if the access modifiers are explicitly stated. I noticed the same issue in some other places too.

kayabuttertoastt commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,78 @@
+                default:
+                    return new InvalidCommand();
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
```
I appreciate the specificity of your error messages!

kayabuttertoastt commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,48 @@
+        try {
+            Task task = taskList.get(taskNumber - 1);
+            task.setTaskState(true);
+        } catch (IndexOutOfBoundsException e) {
```
Perhaps the error message could be more specific by letting the users know if the index is too high or negative. For instance, error messages like "☹ OOPS!!! Task number cannot be negative. Please specify the correct task number.") for negative numbers.

kayabuttertoastt reviewed (commented):
Overall, LGTM, just a few nits.

AnnabelTing committed:
A-JUnit

AnnabelTing committed:
A-JavaDoc

AnnabelTing committed:
A-CodingStandard

AnnabelTing committed:
A-JavaDoc

AnnabelTing committed:
Level-9

AnnabelTing committed:
Merge branch 'branch-A-JavaDoc'

AnnabelTing committed:
Merge branch 'branch-Level-9'

AnnabelTing committed:
A-CheckStyle

miljyy commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -15,7 +16,11 @@ public class ListCommand extends Command {
      */
     public void execute(TaskList tasks, Ui ui, Storage storage) {
-        ui.listTaskResponse(tasks);
+        try {
+            ui.listTaskResponse(tasks);
+        } catch (ChatException e) {
+            ui.showLoadingError(e);
+        }
```
I like how you handled this potential error!

miljyy commented on src/test/java/duke/command/ListCommandTest.java:
```diff
@@ -35,14 +34,15 @@ public void execute_emptyList_success() {
     @Test
```
Perhaps you could indicate which method is being tested?

miljyy reviewed (commented):
Looks good overall!

AnnabelTing committed:
Level-10

AnnabelTing committed:
Add assertions

Assertions are added to document important assumptions that should hold
at various points in the code.

AnnabelTing committed:
Correct style of code

Storage has a long method with several lines of repeated code.

There are some errors in current style of code.

This can affect readability of code.

Let's shorten long method in Storage and correct other style errors.

AnnabelTing committed:
Merge pull request #2 from AnnabelTing/branch-A-Assertions

Add assertions

AnnabelTing committed:
Merge branch 'master' into branch-A-CodeQuality

AnnabelTing committed:
Merge pull request #3 from AnnabelTing/branch-A-CodeQuality

Correct style of code

AnnabelTing committed:
Add MassOps for 'delete', 'mark' and 'unmark' command

The chat bot can only delete, mark or unmark one task item at a time.

This can be very troublesome for users.

Let's allow users to be able to edit multiple tasks at once.

AnnabelTing committed:
Add user guide and edit tests

A user guide was not available and it can be hard for users to navigate.

The tests were not working after some changes so it has been fixed.

AnnabelTing committed:
Update main class

Main class was set to Duke and has been updated to Launcher
